### PR TITLE
Enhance documentation for for SuperTooltip

### PR DESCRIPTION
**Pull request by RotRot-pi**
This pull request improves the documentation for the SuperTooltip widget
by adding more detailed explanations for various parameters and methods.
Key changes include:

* Documentation for parameters like `popupDirection`, `controller`,
  `onShow`, `onHide`, and more.
* Documentation for methods such as `_showTooltip`, `_hideTooltip`,
  `_buildCloseButton`, and `_createOverlayEntries`.

This should make it easier for developers to understand and use the
SuperTooltip widget.

I also have some questions about the following parameters, as I've
tested them but haven't fully grasped their behavior:

* `constraints`
* `verticalOffset`
* `snapsFarAwayHorizontally`
* `snapsFarAwayVertically`

Could you please provide some clarification on these?

Thank you for your time and consideration!

### CODE REVIEW

File: example/android/build.gradle [hunk 1]
```diff
@@ -1,15 +1,3 @@
-buildscript {
-    ext.kotlin_version = '1.6.10'
```
Comment (bensonarafat) [inline example/android/build.gradle:2]: Why did you remove this? @RotRot-pi

File: example/android/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.3-all.zip
```
Comment (bensonarafat) [inline example/android/gradle/wrapper/gradle-wrapper.properties:3]: This?
Can we maintain the previous verison

File: example/android/settings.gradle [hunk 3]
```diff
@@ -1,11 +1,36 @@
+
+
+pluginManagement {
+    def flutterSdkPath = {
+        def properties = new Properties()
```
Comment (bensonarafat) [inline example/android/settings.gradle:5]: this too ?
What version did you set your to?

**Review by bensonarafat — CHANGES_REQUESTED** [review]
Pls can you look at my comments? @RotRot-pi

File: example/android/settings.gradle (hunk 3 above)
Comment (RotRot-pi) [reply example/android/settings.gradle:5]: Hi @bensonarafat,

Thank you for reviewing my pull request.

I made those changes because I'm using Flutter version 3.24, which required updating the Gradle configuration to run the project successfully. Specifically, these changes were necessary due to the deprecated imperative application of Flutter's Gradle plugins. You can find more details in this Flutter article: [Deprecated imperative apply of Flutter's Gradle plugins](https://docs.flutter.dev/release/breaking-changes/flutter-gradle-plugin-apply).

As I'm new to open-source contributions, I appreciate any guidance on best practices. If maintaining the same Gradle version as the original project is preferred, I’m happy to revert the changes or adjust them accordingly.

Thanks again for your feedback and patience!

Comment (bensonarafat) [reply example/android/settings.gradle:5]: It won't be best to increase the flutter version. 
Can we keep it as is for now? Others are using the package. 

We can increase it later, but now is not the best time @RotRot-pi

**Comment by bensonarafat** [comment]
After you do that @RotRot-pi , I will explain the following 

- `constraints`
- `verticalOffset`
- `snapsFarAwayHorizontally`
- `snapsFarAwayVertically`

**Comment by RotRot-pi** [comment]
Should make another PR ?

**Comment by bensonarafat** [comment]
> Should make another PR ?

Can't you correct from here @RotRot-pi ?

**Comment by bensonarafat** [comment]
Hello @RotRot-pi any progress on this

**Comment by RotRot-pi** [comment]
Hello , is there any problems in the documentation ? 
I think i have already solved issue.

**Comment by bensonarafat** [comment]
@RotRot-pi 
I already told you, I can't merge this, downgrade the version
